videostore/utils.go

```diff
@@ -228,42 +258,81 @@ func formatDateTimeToString(dateTime time.Time) string {
 }
 
 // matchStorageToRange returns a list of files that fall within the provided time range.
+// The matcher assumes that the input files list is sorted by start time.
 // Includes trimming video files to the time range if they overlap.
 func matchStorageToRange(files []string, start, end time.Time, logger logging.Logger) []string {
```

Comment:
lets have 2 slices of the same length, one which is the sorted (earliest to latest) valid file names and the other which is the time.Time of those filenames 

That way you don't need to re parse the file names.

Reply:
You could also filter out the last file given frequently it will still be being written to.

`times[0].After(end)` and `times[len(times)-1].Before(from)` and if so bail out early before going in to the loop.

That way you can be confident if you are in the loop, you will be able to find a valid segment and only need to consider the duration within the start and end time.

Reply:
Create new struct [fileWithDate](https://github.com/viam-modules/video-store/pull/53/commits/fd5a2353338d81b13f17e6bdd3d8a292230b2024) to handle mapping. 


Moved away from file end estimate and search for first file to consider beforehand.